code/modules/industrial_lift/tram/tram_lift_master.dm

```diff
@@ -219,3 +228,102 @@
 
 		else
 			stack_trace("Tram doors update_tram_doors called with an improper action ([action]).")
+
+/datum/lift_master/tram/proc/set_operational(new_value)
+	if(is_operational != new_value)
+		is_operational = new_value
+
+/**
+ * Returns the closest tram nav beacon to an atom
+ *
+ * Creates a list of nav beacons in the requested direction
+ * and returns the closest to be passed to the industrial_lift
+ *
+ * Arguments: source: the starting point to find a beacon
+ *            travel_dir: travel direction in tram form, INBOUND or OUTBOUND
+ *            beacon_type: what list of beacons we pull from
+ */
+/datum/lift_master/tram/proc/closest_nav_in_travel_dir(atom/origin, travel_dir, beacon_type)
+	if(!istype(origin) || !origin.z)
+		return FALSE
+
+	var/list/obj/effect/landmark/tram/nav/inbound_candidates = list()
+	var/list/obj/effect/landmark/tram/nav/outbound_candidates = list()
+
+	for(var/obj/effect/landmark/tram/nav/candidate_beacon in GLOB.tram_landmarks[beacon_type])
+		if(candidate_beacon.z != origin.z || candidate_beacon.z != nav_beacon.z)
+			continue
+
+		switch(nav_beacon.dir)
+			if(EAST, WEST)
+				if(candidate_beacon.y != nav_beacon.y)
+					continue
+				else if(candidate_beacon.x < nav_beacon.x)
+					inbound_candidates += candidate_beacon
+				else
+					outbound_candidates += candidate_beacon
+			if(NORTH, SOUTH)
+				if(candidate_beacon.x != nav_beacon.x)
+					continue
+				else if(candidate_beacon.y < nav_beacon.y)
+					inbound_candidates += candidate_beacon
+				else
+					outbound_candidates += candidate_beacon
+
+	switch(travel_dir)
+		if(INBOUND)
+			var/obj/effect/landmark/tram/nav/selected = get_closest_atom(/obj/effect/landmark/tram/nav, inbound_candidates, origin)
+			if(selected)
+				return selected
+			stack_trace("No inbound beacon candidate found for [origin]. Cancelling dispatch.")
+			return FALSE
+
+		if(OUTBOUND)
+			var/obj/effect/landmark/tram/nav/selected = get_closest_atom(/obj/effect/landmark/tram/nav, outbound_candidates, origin)
+			if(selected)
+				return selected
+			stack_trace("No outbound beacon candidate found for [origin]. Cancelling dispatch.")
+			return FALSE
+
+		else
+			stack_trace("Tram receieved invalid travel direction [travel_dir]. Cancelling dispatch.")
+
+	return FALSE
+
+/**
+ * Moves the tram when hit by an immovable rod
+ *
+ * Tells the individual tram parts where to actually go and has an extra safety checks
+ * incase multiple inputs get through, preventing conflicting directions and the tram
+ * literally ripping itself apart. all of the actual movement is handled by SStramprocess
+ *
+ * Arguments: collided_rod (the immovable rod that hit the tram)
```

Comment:
Mention what the return value should be here, for future consumers.

Reply:
Done.